### ISSUE TITLE
Kubernetes pod mismatch in UDP port when trying to create a RTSP session (using ffmpeg)

### ISSUE THREAD

**Issue by rmazzine** [kubernetes/kubernetes#94561]
**What happened**: Kubernetes pod mismatch in UDP port when trying to create a RTSP session (using ffmpeg).

I found a strange behavior in Kubernetes when trying to make a RTSP session (to stream a camera) using ffmpeg. I both tried in a Kubernetes cluster and minikube and they do the same thing.

The full description is bellow:

I enter on a Kubernetes pod with ffmpeg installed and start the RTSP streaming session with the ffmpeg command 

`ffmpeg -i "rtsp://<USER>:<PWD>@<HOST>:<PORT>/<ARGS>" test.mp4`

I also monitor the ports being opened with **netstat -tulpn** and the package communication with **tcpdump**

The strange behavior is the following.

netstat identifies that two UDP ports were opened.

**However the dump from ffmpeg shows it actually tried to access another UDP port, hence it fails instantly (as there's not such port opened).**

And the above **only happend with a container in kubernetes**, the same does not happen in a container outside locally. (therefore, this is why I posted here, it does not look, maybe, to be a problem of ffmpeg)

I also tested both in my external Kubernetes cluster and in minkube (locally), both had the same behavior. 

**What you expected to happen**:

I would expect the created UDP ports by the port being the same as those being accessed my the ffmpeg service (and this happen with a container outside Kubernetes).

**How to reproduce it (as minimally and precisely as possible)**:

(you will need a RTSP connection and a container image with ffmpeg installed)

Create a pod and try to create a RTSP session using ffmpeg (`ffmpeg -i "rtsp://<USER>:<PWD>@<HOST>:<PORT>/<ARGS>" test.mp4`) and use `netstat -tulpn` and `tcpdump` to analyse the ports being created by the pod and the ones that the service (ffmpeg) tries to access.

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`): 1.19
- Cloud provider or hardware configuration: Kubernetes in a VPS server (and also minikube locally)
- OS (e.g: `cat /etc/os-release`): Ubuntu 18.04
- Kernel (e.g. `uname -a`): Linux 4.15.0-112-generic #113-Ubuntu SMP Thu Jul 9 23:41:39 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux
- Install tools: 
- Network plugin and version (if this is a network-related bug):
- Others:

**Comment by athenabot** [comment]
/triage unresolved

Comment `/remove-triage unresolved` when the issue is assessed and confirmed.

🤖 I am a bot run by vllry. 👩‍🔬

**Comment by mattfenwick** [comment]
@rmazzine I'm not super familiar with ffmpeg -- can you elaborate on what your command is supposed to do?  

 - is it trying to send data out of your kube pod to somewhere else?  I wonder if doing something which kubernetes doesn't support
 - is it supposed to create a server for something else to access?  -->> in this case [a service](https://kubernetes.io/docs/concepts/services-networking/service/) should handle it

Can you provide a script to reproduce this behavior?

**Comment by rmazzine** [comment]
Hello @mattfenwick, thansk for your reply!

The specific ffmpeg function I was trying to use was the stream (in this case a video/audio file) from a server and save the stream in the local environment.

Something similar to:

```
ffmpeg -i "rtsp://<HOST>:<PORT>/<PATH>"  <FILENAME_TO_SAVE>
```

I am using that to get a video stream from a DVR (that is connected to some security cameras).

However, I tried to replicate this error with:

- A non-authenticated rtsp stream, which worked (https://www.wowza.com/html/mobile.html here there's an example)

- A authenticated rtsp stream (with this server https://github.com/aler9/rtsp-simple-server and accessing from other completely different computer), which also worked.

Then, I think the problem is actually the DVR server is using (which is from Intelbras but use the API from DAHUA). A weird problem as it only happens with my container in Kubernetes.

However, for those who have the same problem, (and using ffmpeg server wrappers like https://www.npmjs.com/package/node-rtsp-stream), I found another (https://www.npmjs.com/package/node-media-server) that worked fine.

Therefore, thanks again for the support and sorry for the inconvenience! But I hope with this comment I can help some other that can eventually has a similar problem.
